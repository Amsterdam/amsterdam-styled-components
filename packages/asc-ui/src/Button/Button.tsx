import * as React from 'react'
import { Button as ButtonStyled } from '@amsterdam/asc-core'

const Button = (props: any) => (
  <ButtonStyled {...props}>
    {props.text}!!
  </ButtonStyled>
)

export default Button
