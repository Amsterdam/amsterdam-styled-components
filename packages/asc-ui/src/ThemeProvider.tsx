import * as React from 'react'
import { StyledComponents, Theme } from '@datapunt/asc-core'
import defaultTheme from './config/theme'

interface Props {
  theme?: Theme.ThemeInterface,
  children: any
}

const ASCDefaultThemeProvider = ({ theme, children }: Props) => {
  const namespacedTheme = {
    ascTheme: Object.assign({}, defaultTheme.ascTheme, theme),
  }
  return (
    <StyledComponents.ThemeProvider theme={namespacedTheme}>
      {children}
    </StyledComponents.ThemeProvider>
  )
}

export default ASCDefaultThemeProvider
