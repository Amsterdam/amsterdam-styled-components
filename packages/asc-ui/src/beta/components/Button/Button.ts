const Button = () => 'test'

export default Button
