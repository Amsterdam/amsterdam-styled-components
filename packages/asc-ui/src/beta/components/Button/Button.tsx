import type { ElementType, ButtonHTMLAttributes } from 'react'
import { forwardRef } from 'react'
import ButtonStyle from './ButtonStyle'

export type ButtonProps = {
  as?: ElementType
  variant?: 'primary' | 'secondary' | 'tertiary' | 'blank'
  small?: boolean
} & ButtonHTMLAttributes<HTMLButtonElement>

const Button = forwardRef(
  ({ as = 'button', children, ...otherProps }: ButtonProps, ref) => (
    <ButtonStyle as={as} ref={ref} {...otherProps}>
      {children}
    </ButtonStyle>
  ),
)

export default Button
