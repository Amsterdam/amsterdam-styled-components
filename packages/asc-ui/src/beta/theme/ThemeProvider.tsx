import deepMerge from 'deepmerge'
import type { FunctionComponent, ReactNode } from 'react'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascBetaTheme } from '.'
import type { Theme } from '../types'

export interface ThemeProviderProps {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  theme: defaultTheme = ascBetaTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return <AscThemeProvider theme={theme}>{children}</AscThemeProvider>
}

export default ThemeProvider
