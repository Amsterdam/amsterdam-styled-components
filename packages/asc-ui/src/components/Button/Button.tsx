import * as React from 'react'
import ButtonStyle, {
  IconLeft,
  IconRight,
  Props as ButtonStyleProps,
  ButtonVariants,
} from './ButtonStyle'
import Icon, { defaultProps as iconDefaultProps } from '../Icon/Icon'

export type Props = {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  iconLeft?: React.ReactNode
  iconRight?: React.ReactNode
  icon?: React.ReactNode
  iconSize?: number
  $as?: any
} & ButtonStyleProps &
  React.ButtonHTMLAttributes<HTMLButtonElement>

export { ButtonVariants }

const Button: React.FC<Props> = ({
  children,
  iconLeft,
  iconRight,
  icon,
  iconSize,
  $as,
  ...otherProps
}): any => {
  const iconProps = {
    size: iconSize || iconDefaultProps.size,
  }
  return (
    <ButtonStyle {...otherProps} as={$as}>
      {iconLeft && <IconLeft {...iconProps}>{iconLeft}</IconLeft>}
      {icon ? <Icon {...iconProps}>{icon}</Icon> : children}
      {iconRight && <IconRight {...iconProps}>{iconRight}</IconRight>}
    </ButtonStyle>
  )
}

export default Button
