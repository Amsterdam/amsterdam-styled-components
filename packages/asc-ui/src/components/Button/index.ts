export { ButtonProps, default } from './Button'
