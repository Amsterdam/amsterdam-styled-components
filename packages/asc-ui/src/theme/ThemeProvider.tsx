import React, { ReactNode } from 'react'
import deepMerge from 'deepmerge'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascDefaultTheme } from '.'
import { Theme } from '../types'

interface Props {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: React.FC<Props> = ({
  theme: defaultTheme = ascDefaultTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return (
    <AscThemeProvider theme={theme}>
      <>{children}</>
    </AscThemeProvider>
  )
}

export default ThemeProvider
