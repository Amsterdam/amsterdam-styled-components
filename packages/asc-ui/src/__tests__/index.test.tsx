import * as React from 'react'
import { ButtonBarStyled, Theme } from '@amsterdam/asc-core'
import { mount } from 'enzyme'
import ThemeProvider from '../ThemeProvider'
import Button from '../components/Button'
import 'jest-styled-components'

describe('Styled components', () => {
  it('should render buttons', () => {
    const component = mount(
      <ThemeProvider>
        <ButtonBarStyled>
          <Button color={Theme.TypeLevel.primary}>Hello</Button>
          <Button color={Theme.TypeLevel.secondary}>World</Button>
        </ButtonBarStyled>
      </ThemeProvider>,
    )
    expect(component).toMatchSnapshot()
  })
})
