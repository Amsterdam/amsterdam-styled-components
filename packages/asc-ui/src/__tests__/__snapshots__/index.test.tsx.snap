// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled components should render buttons 1`] = `
.c1 {
  border: none;
  color: #fff;
  min-height: 38px;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  background: #004699;
  background-size: 100% 50%;
  background-repeat: no-repeat;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
}

.c1:hover {
  background: #00387a;
  background-size: 100% 50%;
  background-repeat: no-repeat;
}

.c2 {
  border: none;
  color: #fff;
  min-height: 38px;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  background: #ec0000;
  background-size: 100% 50%;
  background-repeat: no-repeat;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
}

.c2:hover {
  background: #bc0000;
  background-size: 100% 50%;
  background-repeat: no-repeat;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
}

.c0 > * {
  margin-right: 5px;
}

<ASCDefaultThemeProvider>
  <ThemeProvider
    theme={
      Object {
        "ascTheme": Object {
          "colors": Object {
            "bright": Object {
              "main": "#ffffff",
            },
            "error": Object {
              "main": "#ec0000",
            },
            "primary": Object {
              "dark": "#00387a",
              "main": "#004699",
            },
            "secondary": Object {
              "dark": "#bc0000",
              "main": "#ec0000",
            },
          },
          "globalStyle": "",
          "typography": Object {
            "primary": "AvenirNextLTW01-Regular, verdana, sans-serif",
            "secondary": "AvenirNextLTW01-Regular, verdana, sans-serif",
          },
        },
      }
    }
  >
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  display: flex;
  padding: 5px;

  & > * {
    margin-right: 5px;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <Button
            color="primary"
          >
            <styled.button
              color="primary"
            >
              <StyledComponent
                color="primary"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "
  border: none;
  color: ",
                        [Function],
                        ";
  min-height: 38px;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  background: ",
                        [Function],
                        ";
  background-size: 100% 50%;
  background-repeat: no-repeat;
  padding: 0 10px 0 10px;
  ",
                        [Function],
                        "
  ",
                        "transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out;",
                        ";

  &:hover {
    background: ",
                        [Function],
                        ";
    background-size: 100% 50%;
    background-repeat: no-repeat;
  }
",
                      ],
                    },
                    "displayName": "styled.button",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <button
                  className="c1"
                  color="primary"
                >
                  Hello
                </button>
              </StyledComponent>
            </styled.button>
          </Button>
          <Button
            color="secondary"
          >
            <styled.button
              color="secondary"
            >
              <StyledComponent
                color="secondary"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "
  border: none;
  color: ",
                        [Function],
                        ";
  min-height: 38px;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  background: ",
                        [Function],
                        ";
  background-size: 100% 50%;
  background-repeat: no-repeat;
  padding: 0 10px 0 10px;
  ",
                        [Function],
                        "
  ",
                        "transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out;",
                        ";

  &:hover {
    background: ",
                        [Function],
                        ";
    background-size: 100% 50%;
    background-repeat: no-repeat;
  }
",
                      ],
                    },
                    "displayName": "styled.button",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <button
                  className="c2"
                  color="secondary"
                >
                  World
                </button>
              </StyledComponent>
            </styled.button>
          </Button>
        </div>
      </StyledComponent>
    </styled.div>
  </ThemeProvider>
</ASCDefaultThemeProvider>
`;
